### PR TITLE
[Feature]: Add onnx evaluation tool

### PR DESCRIPTION
**Pull request by RunningLeon**
This PR includes:

- [x] Add a tool to evaluate ONNX models with ONNXRuntime backend.
- [x] Align metrics of models between PyTorch and ONNXRuntime.
- [x] Update the document.

### Results and Models

<table>
	<tr>
	    <th>Model</th>
	    <th>Config</th>
	    <th>Dataset</th>
	    <th>Metric</th>
	    <th>PyTorch</th>
	    <th>ONNX Runtime</th>
	</tr>
    <tr>
	    <td rowspan="6">ESRGAN</td>
	    <td rowspan="6">esrgan_x4c64b23g32_g1_400k_div2k.py</td>
	    <td rowspan="2">Set5</td>
        <td>PSNR</td>
        <td>28.2700</td>
        <td>28.2619</td>
    </tr>
    <tr>
        <td>SSIM</td>
        <td>0.7778</td>
        <td>0.7784</td>
    </tr>
    <tr>
        <td rowspan="2">Set14</td>
        <td>PSNR</td>
        <td>24.6328</td>
        <td>24.6290</td>
    </tr>
    <tr>
        <td>SSIM</td>
        <td>0.6491</td>
        <td>0.6494</td>
    </tr>
    <tr>
        <td rowspan="2">DIV2K</td>
        <td>PSNR</td>
        <td>26.6531</td>
        <td>26.6532</td>
    </tr>
    <tr>
        <td>SSIM</td>
        <td>0.7340</td>
        <td>0.7340</td>
    </tr>
    <tr>
	    <td rowspan="6">ESRGAN</td>
	    <td rowspan="6">esrgan_psnr_x4c64b23g32_g1_1000k_div2k.py</td>
	    <td rowspan="2">Set5</td>
        <td>PSNR</td>
        <td>30.6428</td>
        <td>30.6307</td>
    </tr>
    <tr>
        <td>SSIM</td>
        <td>0.8559</td>
        <td>0.8565</td>
    </tr>
    <tr>
        <td rowspan="2">Set14</td>
        <td>PSNR</td>
        <td>27.0543</td>
        <td>27.0422</td>
    </tr>
    <tr>
        <td>SSIM</td>
        <td>0.7447</td>
        <td>0.7450</td>
    </tr>
    <tr>
        <td rowspan="2">DIV2K</td>
        <td>PSNR</td>
        <td>29.3354</td>
        <td>29.3354</td>
    </tr>
    <tr>
        <td>SSIM</td>
        <td>0.8263</td>
        <td>0.8263</td>
    </tr>
    <tr>
	    <td rowspan="6">SRCNN</td>
	    <td rowspan="6">srcnn_x4k915_g1_1000k_div2k.py</td>
	    <td rowspan="2">Set5</td>
        <td>PSNR</td>
        <td>28.4316</td>
        <td>28.4120</td>
    </tr>
    <tr>
        <td>SSIM</td>
        <td>0.8099</td>
        <td>0.8106</td>
    </tr>
    <tr>
        <td rowspan="2">Set14</td>
        <td>PSNR</td>
        <td>25.6486</td>
        <td>25.6367</td>
    </tr>
    <tr>
        <td>SSIM</td>
        <td>0.7014</td>
        <td>0.7015</td>
    </tr>
    <tr>
        <td rowspan="2">DIV2K</td>
        <td>PSNR</td>
        <td>27.7460</td>
        <td>27.7460</td>
    </tr>
    <tr>
        <td>SSIM</td>
        <td>0.7854</td>
        <td>0.78543</td>
    </tr>
</table>

**Notes**:

- All ONNX models are evaluated with dynamic shape on the datasets and images are preprocessed according to the original config file.
- This tool is still experimental, and we only support `restorer` for now.

### CODE REVIEW

**Review by Yshuo-Li — COMMENTED** [review]
Is parameter `model_type` necessary since it only supports `restorer`?

File: tools/deploy_test.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    args.save_image = args.save_path is not None
+    model = MMDataParallel(model, device_ids=[0])
+    outputs = single_gpu_test(
+        model,
+        data_loader,
+        save_path=args.save_path,
+        save_image=args.save_image)
+
+    print()
```
Comment (Yshuo-Li) [inline tools/deploy_test.py:73]: `print('')`

Comment (innerlee) [reply tools/deploy_test.py:73]: why

Comment (RunningLeon) [reply tools/deploy_test.py:73]: `print()` is used to print an empty line before printing results.

Comment (Yshuo-Li) [reply tools/deploy_test.py:73]: `print()` is also effective, maybe it is better to be unified with `print('')` in `tools.test.py`.

Comment (Yshuo-Li) [reply tools/deploy_test.py:73]: This does not have to be changed.

**Review by Yshuo-Li — APPROVED** [review]

File: tools/deploy_test.py (hunk 1 above)
Comment (innerlee) [reply tools/deploy_test.py:73]: We can unify all `print('')` in the repo to `print()`, since we prefer consise forms

File: mmedit/core/export/wrappers.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        if save_image:
+            self.base_model.save_image(pred_alpha, meta, save_path, iteration)
+
+        return {'pred_alpha': pred_alpha, 'eval_result': eval_result}
+
+
+class RestorerGenerator(nn.Module):
+
+    def __init__(self, sess, io_binding, output_names):
+        super(RestorerGenerator, self).__init__()
```
Comment (innerlee) [inline mmedit/core/export/wrappers.py:63]: `super().__init__()`

**Comment by innerlee** [comment]
Please add some unit tests for functions and classes in  mmedit/core/export/wrappers.py

**Comment by grimoire** [comment]
@innerlee can we just ignore these functions and classes in Codecov since it is hard to test without a real ONNX model.

**Comment by innerlee** [comment]
> a real ONNX model.

A dummy model contains a single conv or fc is good enough.

**Comment by innerlee** [comment]
If the onnx runtime is not available at CI, you can edit the config similar to this line https://github.com/open-mmlab/mmediting/blob/master/.github/workflows/build.yml#L54
